contracts/PangolinExchange.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.2;
+
+import "@pangolindex/exchange-contracts/contracts/pangolin-periphery/interfaces/IPangolinRouter.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IAssetsExchange.sol";
+import "./SupportedAssets.sol";
+
+/**
+ * @title PangolinExchange
+ * @dev Contract allows user to invest into an ERC20 token
+ * This implementation uses the Pangolin DEX
+ */
+contract PangolinExchange is Ownable, IAssetsExchange {
```

Comment:
Maybe we can change the name to something ore generic, since this contract is just compatible with Uniswap V2 based DEXes

Reply:
Ok, it uses methods with `AVAX` in name so it applies only to Pangolin (right now). Anyway maybe we could mention that Pangolin is compatible with UniswapV2 if anyone would like to check the meaning of used methods